### PR TITLE
Add pre-commit configuration

### PR DESCRIPTION
**Pull request by TechHummel**
Inspired by pre-commit configuration from homeassistant core project.

### CODE REVIEW

File: custom_components/homewhiz/api.py [hunk 1]
```diff
@@ -170,7 +169,7 @@ async def make_api_get_request(
     )
 
     _LOGGER.debug(
-        "Actual canonical request: {0}".format(canonical_request.replace("\n", "\\n"))
+        "Actual canonical request: {}".format(canonical_request.replace("\n", "\\n"))
```
Comment (TechHummel) [inline custom_components/homewhiz/api.py:172]: Not sure what this change is about?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+repos:
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v3.2.2
+    hooks:
+      - id: pyupgrade
```
Comment (rowysock) [inline .pre-commit-config.yaml:5]: there are tools here, that we don't use yet.
We should also add those tools to the github actions, without that we can't enforce that someone used the pre-commit. Maybe there's a way to use the pre-commit config in github actions?

**Comment by rowysock** [comment]
I like this change a lot. I had to fix flake8 so many times after github actions failed!

**Review by rowysock — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 2 above)
Comment (TechHummel) [reply .pre-commit-config.yaml:5]: Either that or I'll copy this from homeassistsnt as well 😀

Comment (rowysock) [reply .pre-commit-config.yaml:5]: it this works, then we should remove the linting from current github actions
https://pre-commit.ci/

**Review by rowysock — APPROVED** [review]

**Pull request merged**